### PR TITLE
Add validation annotations to backend service interfaces

### PR DESCRIPTION
**Pull request by dimas-b**
Also relax test checks to account to differences in validation parameter and method names.

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
please rebase or re-trigger the build. Python failures are been fixed in #5749

**Comment by ajantha-bhat** [comment]
closing and re-opening the PR to re-trigger the build.

File: servers/services/src/main/java/org/projectnessie/services/spi/TreeService.java [hunk 1]
```diff
@@ -41,68 +46,126 @@ public interface TreeService {
...
+          String branchName,
+      @Valid @NotNull @Pattern(regexp = Validation.HASH_REGEX, message = Validation.HASH_MESSAGE)
+          String expectedHash,
       String message,
       List<String> hashesToTransplant,
-      String fromRefName,
+      @Valid
+          @NotBlank
+          @NotNull
+          @Pattern(regexp = Validation.REF_NAME_REGEX, message = Validation.REF_NAME_MESSAGE)
```
Comment (ajantha-bhat) [inline servers/services/src/main/java/org/projectnessie/services/spi/TreeService.java:119]: a) does `REF_NAME_REGEX` handle the blanks too? Do we need this to be `NotBlank` or just `NotNull` is enough?

b) do we need validations for `hashesToTransplant` too?

Comment (dimas-b) [reply servers/services/src/main/java/org/projectnessie/services/spi/TreeService.java:119]: I used the same validation rules as in v1 `HttpTreeApi` and related classes.

`NotBlank` may actually produce a more intuitive error message,

Comment (dimas-b) [reply servers/services/src/main/java/org/projectnessie/services/spi/TreeService.java:119]: We could validate `hashesToTransplant`, but we do not do that in API v1 and did not want to change behaviour in this PR.

Comment (ajantha-bhat) [reply servers/services/src/main/java/org/projectnessie/services/spi/TreeService.java:119]: I thought it is freshly added validations. Ok to be inline with v1 as of now and improve in the follow up.

**Review by ajantha-bhat — APPROVED** [review]

**Pull request merged**